specification/communication/Communication.Messages/models.tsp

```diff
@@ -0,0 +1,370 @@
+import "@typespec/rest";
+import "@typespec/http";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+using Azure.Core.Traits;
+using Azure.ClientGenerator.Core;
+
+@versioned(AzureCommunicationMessagesService.Versions)
+namespace AzureCommunicationMessagesService;
+
+@access(Access.public)
+@doc("The type of message. Supports text, image, template.")
+enum CommunicationMessageType {
+  @doc("Text message type.")
+  text,
+
+  @doc("Image message type.")
+  image,
+
+  @doc("Template message type.")
+  template,
+}
+
+@access(Access.public)
+@doc("The WhatsApp button sub type.")
+enum WhatsAppMessageButtonSubType {
+  @doc("The WhatsApp button sub type is quick reply.")
+  quickReply,
+
+  @doc("The WhatsApp button sub type is url.")
+  url,
+}
+
+@access(Access.public)
+@doc("The aggregated template status.")
+enum MessageTemplateStatus {
+  @doc("Message template is approved.")
+  approved,
+
+  @doc("Message template is rejected.")
+  rejected,
+
+  @doc("Message template is pending.")
+  pending,
+
+  @doc("Message template is paused.")
+  paused,
+}
+
+@doc("The type of the template parameter.")
+enum MessageTemplateValueType {
+  @doc("The text template parameter type.")
+  text,
+
+  @doc("The image template parameter type.")
+  image,
+
+  @doc("The document template parameter type.")
+  document,
+
+  @doc("The video template parameter type.")
+  video,
+
+  @doc("The location template parameter type.")
+  location,
+
+  @doc("The quick action template parameter type.")
+  quickAction,
+}
+
+@doc("The type of the communication messages channel.")
+enum CommunicationMessagesChannel {
+  @doc("The WhatsApp communication messages channel type.")
+  whatsApp,
+}
+
+@doc("The type of the message template.")
+enum MessageTemplateBindingsType {
```

Comment:
We prefer "kind" over "type" for discriminators.

Reply:
Same for any enums or model property names throughout these specs.

Reply:
Changed to `Kind`